### PR TITLE
Basal picker

### PR DESCRIPTION
**Pull request by ps2**

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1 +1 @@
-github "LoopKit/LoopKit" "dev"
+github "tidepool-org/LoopKit" "basal-picker"
```
Comment (mpangburn) [inline Cartfile:1]: What's your typical Cartfile strategy for merging changes that depend on different branches? Switch to the dependent branch while testing, then revert the Cartfile after merging?

**Review by mpangburn — COMMENTED** [review]
LGTM! 🚀

File: MinimedKit/Models/PumpModel.swift [hunk 2]
```diff
@@ -99,6 +99,29 @@ public enum PumpModel: String {
     var usesTwoBytesForMaxBolus: Bool {
         return generation >= 23
     }
+
+    public var supportedBasalRates: [Double] {
+        if generation >= 23 {
+            // 0.025 units (for rates between 0.0-0.975 U/h)
+            let rateGroup1 = ((0...38).map { Double($0) / Double(pulsesPerUnit) })
```
Comment (darinkrauss) [inline MinimedKit/Models/PumpModel.swift:106]: Should `38` be `39`?

File: Cartfile (hunk 1 above)
Comment (ps2) [reply Cartfile:1]: Yes. For me, I can build in a workspace setup, and not worry to much about updating carthage specs, but travis needs it updated, and if others check out the branch, it helps them be able to build.

File: MinimedKit/Models/PumpModel.swift [hunk 3]
```diff
@@ -99,6 +99,29 @@ public enum PumpModel: String {
...
+            let rateGroup3 = ((100...350).map { Double($0) / Double(pulsesPerUnit/4) })
+            return rateGroup1 + rateGroup2 + rateGroup3
+        } else {
+            // 0.05 units for rates between 0.0-35U/hr
+            return (0...700).map { Double($0) / Double(pulsesPerUnit) }
+        }
+    }
+
+    public var maximumBasalScheduleEntryCount: Int {
+        return 48
```
Comment (darinkrauss) [inline MinimedKit/Models/PumpModel.swift:119]: Really? Wow! I didn't know it supported 48.

File: MinimedKit/Models/PumpModel.swift (hunk 2 above)
Comment (darinkrauss) [reply MinimedKit/Models/PumpModel.swift:106]: @ps2 Here.

**Review by darinkrauss — COMMENTED** [review]
I think the range for `rateGroup1` is incorrect and the `Cartfile` and `Cartfile.resolved` need to be fixed to point back to `LoopKit/LoopKit`, but otherwise looks good.

File: MinimedKit/Models/PumpModel.swift (hunk 2 above)
Comment (ps2) [reply MinimedKit/Models/PumpModel.swift:106]: Thanks.  Fixed here: https://github.com/ps2/rileylink_ios/commit/23af71639041ca3a2a1b4120d3cd0a06c5d3fa01

File: MinimedKit/Models/PumpModel.swift (hunk 3 above)
Comment (ps2) [reply MinimedKit/Models/PumpModel.swift:119]: Yeah, It was a little bit of a pain to test, but the pump accepted it!

**Pull request merged**